### ISSUE TITLE
MethodError calling jl_show from embedding API

### ISSUE THREAD

**Issue by ihnorton** [JuliaLang/julia#20580]
This worked in 0.5. Reduced from `examples/embedding.c`.

(see https://github.com/JuliaLang/julia/pull/20535 for build fixes if needed)

```
int main()
{
    jl_init(JULIA_INIT_DIR);

    jl_value_t* rv = jl_eval_string("UndefVarError(:foo)");

    jl_show(jl_stderr_obj(), rv);

    int ret = 0;
    jl_atexit_hook(ret);
    return ret;
}
```

```
fatal: error thrown and no exception handler available.
Base.MethodError(f=Base.#show(), args=(Base.TTY(handle=0x00007f9d29c54cd0, status=3, buffer=Base.AbstractIOBuffer{Array{UInt8, 1}}(data=Array{UInt8, 1}[], readable=true, writable=true, seekable=false, append=true, size=0, maxsize=9223372036854775807, ptr=1, mark=-1), readnotify=Base.Condition(waitq=Array{Any, 1}[]), closenotify=Base.Condition(waitq=Array{Any, 1}[]), sendbuf=Base.Nullable{Base.AbstractIOBuffer{Array{UInt8, 1}}}(hasvalue=false, value=#<null>), lock=Base.ReentrantLock(locked_by=Base.Nullable{Task}(hasvalue=false, value=#<null>), cond_wait=Base.Condition(waitq=Array{Any, 1}[]), reentrancy_cnt=0), throttle=10485760), UndefVarError(var=:foo)), world=0x0000000000000000)
rec_backtrace at /Users/inorton/git/julia/usr/lib/libjulia.dylib (unknown line)
jl_throw at /Users/inorton/git/julia/usr/lib/libjulia.dylib (unknown line)
jl_method_error_bare at /Users/inorton/git/julia/usr/lib/libjulia.dylib (unknown line)
jl_method_error at /Users/inorton/git/julia/usr/lib/libjulia.dylib (unknown line)
jl_apply_generic at /Users/inorton/git/julia/usr/lib/libjulia.dylib (unknown line)
jl_show at /Users/inorton/git/julia/usr/lib/libjulia.dylib (unknown line)
main at /Users/inorton/git/julia/examples/./embedding (unknown line)
```

**Comment by vtjnash** [comment]
Looks like `jl_show` needs to be updated for #265 fix (or just gotten rid of)

**Comment by JeffBezanson** [comment]
Fixed by #21299
